src/sonic_ax_impl/mibs/vendor/cisco/ciscoSwitchQosMIB.py

```diff
@@ -110,8 +106,17 @@ def update_stats(self):
         self.mib_oid_to_queue_map = {}
         self.mib_oid_list = []
 
-        for if_index in self.if_range:
-            for queue in self.port_queue_list_map[if_index]:
+        # Update queue counters for port
+        if_range = sorted(list(self.oid_sai_map.keys()))
+        for if_index in if_range:
+            if_queues = []
+            try:
+                if_queues = self.port_queue_list_map[if_index]
+            except KeyError:
```

Comment:
>KeyError [](start = 19, length = 8)

Does it indicate DB corruption? If yes, log it seriously.

Reply:
No, It doesn't mean the  DB corruption. 
It means that port does not have configured queue counters. 
The appropriate comment is already added.

Reply:
I see. Suggest use non-exception code to check, because it is a quite common use case. Exception is generally to handle exceptional cases.

---
In reply to: [158642470](https://github.com/andrii-savka/sonic-snmpagent/pull/5#discussion_r158642470) [](ancestors = 158642470)

Reply:
ok. I will change it to a non exceptional verification.

Reply:
Done.